### PR TITLE
NAVAND-1102: add APIs to pause and resume route refreshes

### PR DESCRIPTION
**Pull request by dzinad**

### CODE REVIEW

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshPauseResumeIntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package com.mapbox.navigation.core.routerefresh
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class, ExperimentalPreviewMapboxNavigationAPI::class)
+internal class RouteRefreshPauseResumeIntegrationTest : RouteRefreshIntegrationTest() {
```
Comment (VysotskiVadim) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshPauseResumeIntegrationTest.kt:9]: what do you think about testing the case, when resume called multiple times?

**Review by VysotskiVadim — APPROVED** [review]

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshPauseResumeIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Test
+    fun pause_and_resume_refreshes() = coroutineRule.runBlockingTest {
+        val routes = setUpRoutes("route_response_single_route_refresh.json")
+        routeRefreshController = createRefreshController(60_000)
```
[the remark this answers is not in the document]
Comment (dzinad) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshPauseResumeIntegrationTest.kt:14]: Yes. There's only one argument.

File: instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/coroutines/Adapters.kt [hunk 3]
```diff
@@ -182,3 +184,16 @@ fun <T, OBSERVER> loggedCallbackFlow(
...
+}
+
+suspend fun <T> List<T>.waitUntilCondition(condition: (List<T>) -> Boolean) {
+    val list = this
+    withTimeout(10000) {
+        while (!condition(list)) {
+            delay(50)
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (dzinad) [reply instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/coroutines/Adapters.kt:199]: Not anymore.

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshPauseResumeIntegrationTest.kt (hunk 1 above)
Comment (dzinad) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshPauseResumeIntegrationTest.kt:9]: It's covered in `PlannedRouteRefreshControllerTest`. I'm not sure whether adding an integration test too would make a lot of sense here.

[the remark this answers is not in the document]
Comment (dzinad) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshPauseResumeIntegrationTest.kt:9]: Added.

**Pull request merged**